pennylane/math/fn.py

```diff
@@ -339,6 +492,39 @@ def get_interface(tensor):
     return TensorBox(tensor).interface
 
 
+def marginal_prob(prob, axis):
+    """Compute the marginal probability given joint probabilities of the full system.

```

Comment:
Hm, the implementation makes sense for quantum probabilities, but the function is named as if it is agnostic of how the probability came about....rename to "marginal_quantum_prob"? Ok, that looks ugly...Or maybe this:
```suggestion
    """Compute the marginal probability given a joint probability distribution expressed as a tensor.
    Each random variable corresponds to a dimension. 
    
    If the distribution arises from a quantum circuit measured in computational basis, each dimension 
    corresponds to a wire. For example, for a 2-qubit quantum circuit `prob[0, 1]` is the probability of measuring the 
    first qubit in state 0 and the second in state 1. 
```